### PR TITLE
fix|Support screenshot in Area #136

### LINKED ISSUE #136: Support screenshot in Area
When using `area.get` or `area.check` with `vision=True`, the screenshot of the whole page/screen is taken. To have more stable results and pass less data to LLMs, the screenshot of the area itself should be taken.

[See comment](https://github.com/alumnium-hq/alumnium/pull/133/files#r2232781747)

### PR DESCRIPTION
**Pull request by VinayakaMayura**
# Pull Request Template

Thank you for your contribution to Alumnium!
Please fill out this template to help us understand and review your PR efficiently.

## Summary
What is this pull request about?
this is a fix for issue #136 

## Motivation
Why is this change being made?
It captures screenshot of the particular element instead of the whole page

## Related Issue(s)
If this PR fixes or is related to an open issue, link it here.
Example: Closes #136

Closes #

## Type of Change
Mark the relevant options.

-  Bug fix  
- [ ] New feature (non-breaking changes, test coverage, refactoring)
- [ ] Breaking change (fix, refactoring, or feature that would cause existing functionality to change)
- [ ] Cleanup (documentation, formatting)

## Code or UI Demos (if applicable)
- Ran `poetry run pytest` in local and all the tests are passing. Will paste log if required

## Checklist
Please verify the following before submitting:

- [ ] I have read the [Contributing Guidelines](https://github.com/alumnium-hq/alumnium/blob/main/CONTRIBUTING.md)
- [ ] I have added or updated relevant documentation
- [ ] I have written or updated necessary tests
- [ ] I have tested the changes locally
- [ ] The changes follow the project's coding style and structure
- [ ] I have not included any sensitive or credential-related information
- [ ] I have ensured these changes maintain or improve accessibility (for UI changes)
- [ ] I have considered security implications of these changes

## Additional Notes
Anything else the reviewer should be aware of?
For example: limitations, discussions to follow up on, or next steps.

### CODE REVIEW

**Comment by p0deje** [comment]
Tried to test this locally in Selenium on Google and ran into some errors.

```py
>>> al.area("search box").check("has google logo", vision=True)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/p0deje/Development/alumnium/alumnium/area.py", line 66, in check
    screenshot=self.driver.area_screenshot(self.id) if vision else None,
  File "/Users/p0deje/Development/alumnium/alumnium/drivers/selenium_driver.py", line 71, in area_screenshot
    return element.screenshot_as_base64
  File "/Users/p0deje/Development/alumnium/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py", line 315, in screenshot_as_base64
    return self._execute(Command.ELEMENT_SCREENSHOT)["value"]
  File "/Users/p0deje/Development/alumnium/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py", line 395, in _execute
    return self._parent.execute(command, params)
  File "/Users/p0deje/Development/alumnium/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py", line 384, in execute
    self.error_handler.check_response(response)
  File "/Users/p0deje/Development/alumnium/.venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py", line 232, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.WebDriverException: Message: unknown error: unhandled inspector error: {"code":-32000,"message":"Cannot take screenshot with 0 width."}
  (Session info: chrome=138.0.7204.184)
```

**Comment by VinayakaMayura** [comment]
I also added a test but it is failing with different reasons(id not found, width of the element is zero), couldn't figure out what is the exact issue.

I am not sure if the approach we are trying is it correct because by the time we fetch area of the element, the  accessability tree the dom strucutre would have changed

**Comment by VinayakaMayura** [comment]
I was able to make the below test case pass

`def test_area_screenshot_with_vision(al, navigate):
    navigate("https://seleniumbase.io/apps/calculator")
    al.do("2 + 3 =")
    al.check("output of the calculator app should be 5", vision=True)
    al.do("18 - 8 =")
    al.check("output of the calculator app should be 10", vision=True)`

the `do` method in appium class has a retry of 2 by default and it is causing issue. correct me if i am wrong 

`
examples/pytest/area_screenshot_test.py .                                                                                       [100%]

==================================================== 1 passed, 8 warnings in 7.43s ====================================================

 ~/Doc/c/public/alumnium  main !3 ?1 `

**Comment by p0deje** [comment]
```
FAILED examples/pytest/area_screenshot_test.py::test_area_screenshot_with_vision - AssertionError: The screenshot shows the output of the calculator app as 'Error', which indicates that the output is not 5. Therefore, the statement that the output should be 5 is false.
FAILED examples/pytest/swag_labs_test.py::test_checkout - AttributeError: 'Alumni' object has no attribute 'id'
```

There still seem to be some problems with this approach.

File: alumnium/alumni.py [hunk 1]
```diff
@@ -121,7 +121,7 @@ def check(self, statement: str, vision: bool = False) -> str:
             self.driver.accessibility_tree.to_xml(),
             title=self.driver.title,
             url=self.driver.url,
-            screenshot=self.driver.screenshot if vision else None,
+            screenshot=self.driver.area_screenshot(self.id) if vision else None,
```
Comment (p0deje) [inline alumnium/alumni.py]: We should not do that, taking screenshot on `Alumni` we should take the whole page.

Comment (VinayakaMayura) [reply alumnium/alumni.py]: Removed this and added a default in area_screenshot.py
